libmy/my_num_to_str.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023 by Farsight Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "my_num_to_str.h"
+
+static const char *hexchars = "0123456789abcdef";
+static const char *HEXCHARS = "0123456789ABCDEF";
+
+
+static inline const char *
+_my_uint8_to_hex_str_padded(uint8_t num, const char * src, char *dst) {
+	dst[0] = src[(num >> 4) & 0xf];
+	dst[1] = src[num & 0xf];
+	dst[2] = '\0';
+	return dst;
+}
+const char *
+my_uint8_to_hex_str_padded(uint8_t num, char *dst)
+{
+	return _my_uint8_to_hex_str_padded(num, hexchars, dst);
+}
+
+const char *
+my_uint8_to_hex_STR_padded(uint8_t num, char *dst)
+{
+	return _my_uint8_to_hex_str_padded(num, HEXCHARS, dst);
+}
+
+const char *
+my_uint16_to_hex_str(uint16_t num, char *dst)
```

Comment:
Numerical conversion to hex is a fundamentally different (and simpler) problem than conversion to base 10. This routine is as simple as converting to big endian with htons() and emitting the two hex character representation of each byte in a fixed size array.

Reply:
What is your suggestion ? Remove them ? Move them to another file as they are not related to conversion ?